goGene/goGene.go

```diff
@@ -0,0 +1,137 @@
+package goGene
+
+import (
+	"errors"
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/vcf"
+)
+
+type Feature int32
+type MutationType byte
+
+const (
+	Intron   Feature = -1
+	UtrThree Feature = -3
+	UtrFive  Feature = -5
+	// All positive values refer to cDNA position
+
+	Silent     MutationType = 0
+	Missense   MutationType = 1
+	Nonsense   MutationType = 2
+	Frameshift MutationType = 3
+	Intergenic MutationType = 4
+	Intronic   MutationType = 5
+	Splice     MutationType = 6 // +/- 1-2 of E-I boundary
+	FarSplice  MutationType = 7 // +/- 3-10 of E-I boundary
+)
+
+// GoGene is a processed version of a gtf record that enables easy
+// traversal and manipulation of genes on the genomic and mRNA levels.
+type GoGene struct {
+	id           string
+	startPos     int
+	strand       bool
+	cdsStarts    []int
+	genomeSeq    []dna.Base
+	cdnaSeq      []dna.Base
+	featureArray []Feature
+}
```

Comment:
From looking ahead in the code I see that genomeSeq points into the fasta sequence, while cdnaSeq has its own memory allocated.  If I am understanding everything correctly, that means that genomeSeq is memory efficient since it is really only a pointer into the fasta sequence.  I think the downside of this could be some difficult to trackdown bugs and possibly mysterious behavior.  For example, if someone calls myGene.genomeSeq[100] = dna.A, they will not only be setting that base to A in the gene "myGene", but will also be editing the genome and editing that base position in any overlapping isoforms or genes (but not changing any cdnaSeq).  I wonder if storing a transcription end (or transcription length) would give you the same info as having genomeSeq, but make it more clear that the genome sequence was only being stored once (the fasta file).

Reply:
I thought that reslicing to a new variable created a copy rather than a pointer, but after testing in Go Playground I see that you are right. My intention was to make a copy so the sequence could be easily changed. I have switch to using `copy`